### PR TITLE
RealMetricName provides a toString implementation

### PR DESCRIPTION
**Pull request by carterkozak**
## Before this PR
`RealMetricName@012abc`

## After this PR
==COMMIT_MSG==
RealMetricName provides a toString implementation matching ImmutableMetricName.
==COMMIT_MSG==

### CODE REVIEW

File: tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java [hunk 1]
```diff
@@ -36,11 +36,13 @@ public void compareSame() {
...
+        MetricName three = RealMetricName.create(two);
 
-        assertThat(one).isEqualTo(two);
-        assertThat(two).isEqualTo(one);
+        assertThat(one).isEqualTo(two).isEqualTo(three);
+        assertThat(two).isEqualTo(one).isEqualTo(three);
+        assertThat(three).isEqualTo(one).isEqualTo(two);
 
-        assertThat(one.toString()).isEqualTo(two.toString());
+        assertThat(one.toString()).isEqualTo(two.toString()).isEqualTo(three.toString());
```
Comment (schlosna) [inline tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java]: minor nit: do we want to verify the actual contents of the `toString` (i.e. there's an unlikely alternate universe where `one == two == three` if we had a flyweight cache behind builder & RealMetricName)?

```suggestion
        assertThat(one.toString())
                .isEqualTo(two.toString())
                .isEqualTo(three.toString())
                .isEqualTo("MetricName{safeName=test, safeTags={key=value, key1=value1, key2=value2}}");
```

File: tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java [hunk 2]
```diff
@@ -36,11 +36,13 @@ public void compareSame() {
                 .putSafeTags("key", "value")
                 .putSafeTags("key1", "value1")
                 .build();
+        MetricName three = RealMetricName.create(two);
 
-        assertThat(one).isEqualTo(two);
-        assertThat(two).isEqualTo(one);
+        assertThat(one).isEqualTo(two).isEqualTo(three);
+        assertThat(two).isEqualTo(one).isEqualTo(three);
+        assertThat(three).isEqualTo(one).isEqualTo(two);
```
Comment (schlosna) [inline tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java]: Does assertj have any equivalent to perform the full equals/hashCode comparison reflexive, symmetric, transitive permutations like `com.google.common.testing.EqualsTester` [`org.assertj.core.test.EqualsHashCodeContractAssert`](https://github.com/joel-costigliola/assertj-core/blob/master/src/test/java/org/assertj/core/test/EqualsHashCodeContractAssert.java) seems to be what we'd want, but not sure why its not in main assertj module.

Might also want to verify `hashCode` while we're at it.
```
        MetricName four = ImmutableMetricName.copyOf(three);

        assertThat(one)
                .hasSameHashCodeAs(two)
                .hasSameHashCodeAs(three)
                .hasSameHashCodeAs(four);
```

File: tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/RealMetricName.java [hunk 3]
```diff
@@ -50,6 +50,11 @@ public String safeName() {
         return safeTags;
     }
 
+    @Override
+    public String toString() {
+        return "MetricName{safeName=" + safeName + ", safeTags=" +  safeTags + '}';
```
Comment (schlosna) [inline tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/RealMetricName.java:55]: should we condense this, just make it JSON (like) for simplicity?

File: tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java (hunk 1 above)
Comment (carterkozak) [reply tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java]: good call

File: tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/RealMetricName.java (hunk 3 above)
Comment (carterkozak) [reply tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/RealMetricName.java:55]: This matches ImmutableMetricName, I'd prefer to have RealMetricName match the other implementation for now, I'm not sure there's enough benefit to changing the structure to something json-like for it to be worth the change. Thoughts?

Comment (schlosna) [reply tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/RealMetricName.java:55]: as-is is fine by me

File: tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java (hunk 2 above)
Comment (carterkozak) [reply tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java]: > Does assertj have any equivalent to perform the full equals/hashCode comparison reflexive, symmetric, transitive permutations

Not that I'm aware of, I can file an assertj FR upstream though.

> Might also want to verify hashCode while we're at it.

Good idea, done.

**Review by schlosna — APPROVED** [review]

**Pull request merged**